pandevice/userid.py

```diff
@@ -299,6 +299,19 @@ def get_registered_ip(self, ip=None, tags=None, prefix=None):
             dict: ip addresses as keys with tags as values
 
         """
+        start = 1
+        limit = 500
+        addresses = {}
+        while True:
+            newaddr = self.get_registered_ip_loop(ip, tags, prefix, start, limit)
+            rescnt = len(newaddr.keys())
+            addresses.update(newaddr)
+            if (rescnt < limit):
+                break
+            start += limit
+        return addresses
+
+    def get_registered_ip_loop(self, ip=None, tags=None, prefix=None, start=1, limit=500):
```

Comment:
Given the overall updates you'll need to make, I think it makes more sense to keep this as a single function, but if you want it to be two functions, then please make this `_get_registered_ip_loop()` instead.

Reply:
Splitting it in two functions made the changes visibly small. I'm fine if it would be a single function again.